### PR TITLE
Skip update manager check when higher version product is already installed on the system

### PR DESCRIPTION
**Pull request by sharadkjaiswal**
- Looks into registry to find all the installed products on the system
- If the latest product version is higher than the current product version, it will skip product update check.
- Fixes http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6756

**PTAL**
- [ ] @aparajit-pratap

### CODE REVIEW

File: src/DynamoCore/UpdateManager/UpdateManager.cs [hunk 1]
```diff
@@ -1023,8 +1023,62 @@ void client_DownloadProgressChanged(object sender, DownloadProgressChangedEventA
         /// </summary>
         internal static void CheckForProductUpdate()
         {
+            //If we already have higher version installed, don't look for product update.
+            if(new DynamoLookUp().LatestProduct > Instance.ProductVersion)
```
Comment (aparajit-pratap) [inline src/DynamoCore/UpdateManager/UpdateManager.cs]: What's the advantage of doing this over calling it as a static method?

**Comment by aparajit-pratap** [comment]
@sharadkjaiswal do not these API's to get all Dynamo installations and getting the latest product already exist in your `ProductLookup` class. Other than this, LGTM

File: src/DynamoCore/UpdateManager/UpdateManager.cs (hunk 1 above)
Comment (sharadkjaiswal) [inline src/DynamoCore/UpdateManager/UpdateManager.cs]: In order to mock the registry specific calls, I have made those methods as virtual and hence I need an instance method/property to call those virtual methods that are mocked in test environment.

**Comment by sharadkjaiswal** [comment]
Thanks @aparajit-pratap the ProductLookup is sitting on a different module and for the moment I am not looking for creating a dependency on that module. Here my needs is simple, so just implemented that simple logic here.

**Pull request merged**